src/main/scala/edu/ie3/simona/sim/setup/ExtSimLoader.scala

```diff
@@ -52,19 +52,11 @@ object ExtSimLoader extends LazyLogging {
       }
   }
 
-  def loadExtLink(myJar: File): ExtLinkInterface = {
-    val classLoader = new URLClassLoader(
-      Array(myJar.toURI.toURL),
-      this.getClass.getClassLoader
-    )
-    val classToLoad = Class.forName(extLinkClassPath, true, classLoader)
-    classToLoad.getDeclaredConstructor().newInstance() match {
-      case extSim: ExtLinkInterface =>
-        extSim
-      case other =>
-        throw new ClassCastException(
-          s"$extLinkClassPath in loaded jar ${myJar.getPath} is of wrong type ${other.getClass.getSimpleName}"
-        )
-    }
+  def loadExtLink(myJar: File): Option[ExtLinkInterface] = {
+    val classLoader = new URLClassLoader(Array(myJar.toURI.toURL))
+    ServiceLoader
+      .load(classOf[ExtLinkInterface], classLoader)
+      .asScala
+      .headOption
```

Comment:
Now that I think of it, we should log a warning here if more than one implementation of `ExtLinkInterface` are loaded, since every jar should only contain exactly one.

Furthermore, we should log a warning if there's 0 classes loaded.

Both these checks could save the user a lot of debugging in the future.

Reply:
Added a logger that logs an the loading of the external simulation.  If one implementation of ``ExtLinkInterface`` is loaded, an info is logged, if more or 0 implementations are loaded, a warning is logged.